src/olympia/core/db/base.py

```diff
@@ -0,0 +1,49 @@
+from django.db.backends.mysql.base import (
+    CursorWrapper as MySQLCursorWrapper,
+    DatabaseWrapper as MySQLDBWrapper,
+    DatabaseIntrospection as MySQLDBIntrospection,
+    DatabaseSchemaEditor as MySQLDBSchemeEditor)
+
+
+class CursorWrapper(MySQLCursorWrapper):
+    def execute(self, query, args=None):
+        return super(CursorWrapper, self).execute(query, args)
+
+
+class DatabaseIntrospection(MySQLDBIntrospection):
+    def get_field_type(self, data_type, description):
+        field_type = super().get_field_type(data_type, description)
+        if 'auto_increment' in description.extra:
+            if field_type == 'IntegerField':
+                if description.is_unsigned:
+                    return 'PositiveAutoField'
+        return field_type
+
+
+class DatabaseSchemaEditor(MySQLDBSchemeEditor):
+    def create_model(self, model):
+        for field in model._meta.local_fields:
+            # Autoincrement SQL
+            if field.get_internal_type() == "PositiveAutoField":
+                autoinc_sql = self.connection.ops.autoinc_sql(
+                    model._meta.db_table, field.column)
```

Comment:
I'm not sure if all this is actually needed, shouldn't we be able to do that via the field already?

```python
class UnsignedAutoField(models.AutoField):
    def db_type(self, connection):
        return 'integer UNSIGNED AUTO_INCREMENT'

    def rel_db_type(self, connection):
        return 'integer UNSIGNED'
```

(untested… something like that?)

Reply:
It should be unnecessary but there's seemingly some edge case where you can indicate auto-incrementing later. I'm tempted to delete it.  

>
```python
class UnsignedAutoField(models.AutoField):
    def db_type(self, connection):
        return 'integer UNSIGNED AUTO_INCREMENT'

    def rel_db_type(self, connection):
        return 'integer UNSIGNED'
```

We don't need to do this normally - the code in `Field` that builds the sql for the column adds on the `AUTO_INCREMENT`, etc.